contract/src/stage1/function_call.rs

```diff
@@ -23,50 +25,66 @@ impl Keypom {
-                account_id_field: method.account_id_field.clone(),
-                drop_id_field: method.drop_id_field.clone(),
-                key_id_field: method.key_id_field.clone(),
-                funder_id_field: method.funder_id_field.clone(),
-            };
-
+        for (i, method) in methods.iter().enumerate() {
             let mut final_args = method.args.clone();
-
-            if final_args.contains("\"keypom_args\"") {
+            if final_args.len() == 0 {
+                final_args = "{}".to_string();
+            }
+            
+            let try_json: Result<Value, _> = from_str(&final_args);
+            if try_json.is_err() {
                 near_sdk::log!(
-                    "Keypom Args detected in client args. Returning and decrementing keys"
+                    "Cannot cast args to JSON. Returning and decrementing keys"
                 );
                 return;
             }
 
-            if final_args.len() == 0 {
-                final_args = format!(
-                    "{{\"keypom_args\":{}}}",
-                    near_sdk::serde_json::to_string(&keypom_args).unwrap()
-                );
-            } else {
-                final_args.insert_str(
-                    final_args.len() - 1,
-                    &format!(
-                        ",\"keypom_args\":{}",
-                        near_sdk::serde_json::to_string(&keypom_args).unwrap()
-                    ),
-                );
+            let mut final_args_json = try_json.unwrap();
+
+            // Check if user provided args is present and fc_args is some
+            if let (Some(rule), Some(user_args_str)) = (method.user_args_rule.as_ref(), fc_args.clone().and_then(|a| a[i].clone())) {
+                let try_user_json: Result<Value, _> = from_str(&user_args_str);
+                if try_user_json.is_ok() {
+                    let mut user_args_json = try_user_json.unwrap();
+                    
+                    match rule {
+                        UserArgsRule::AllUser => {
+                            final_args_json = user_args_json;
+                        }
+                        UserArgsRule::FunderPreferred => {
+                            // Take the final args string and merge the user args into it and overwrite any duplicate keys
+                            merge_json(&mut user_args_json, &final_args_json);
+                            final_args_json = user_args_json;
+                        }
+                        UserArgsRule::UserPreferred => {
+                            // Take the final args string and merge the user args into it and overwrite any duplicate keys
+                            merge_json(&mut final_args_json, &user_args_json);
+                        }
+                    }
+                } else {
+                    near_sdk::log!(
+                        "Cannot cast user provided args to JSON. Disregarding user args and continuing"
```

Comment:
is this expected behavior? Seems like something we're choosing / should warn the user about?

I think we should do the same as above, nothing. IMHO.

{"myProposal":"something } in the wrong format"}

uh oh! proposal created with no actual body... key deleted, user mad...

Reply:
We could skip the function call, yea - let me do that.